src/zenml/integrations/label_studio/steps/label_studio_standard_steps.py

```diff
@@ -0,0 +1,234 @@
+
+logger = get_logger(__name__)
+
+
+class LabelStudioDatasetRegistrationConfig(BaseStepConfig):
+    """Step config  when registering a dataset with Label Studio.
+
+    Attributes:
+        label_config: The label config to use for the annotation interface.
+        dataset_name: Name of the dataset to register.
+    """
+
+    label_config: str
+    dataset_name: str
+
+
+class LabelStudioDatasetSyncConfig(BaseStepConfig):
+    """Step config when syncing data to Label Studio.
+
+    Attributes:
+        storage_type: The type of storage to sync to.
+        label_config_type: The type of label config to use.
+
+        prefix: Specify the prefix within the cloud store to import your data
+            from.
+        regex_filter: Specify a regex filter to filter the files to import.
+        use_blob_urls: Specify whether your data is raw image or video data, or
+            JSON tasks.
+        presign: Specify whether or not to create presigned URLs.
+        presign_ttl: Specify how long to keep presigned URLs active.
+        description: Specify a description for the dataset.
+
+        azure_account_name: Specify the Azure account name to use for the
+            storage.
+        azure_account_key: Specify the Azure account key to use for the
+            storage.
+        google_application_credentials: Specify the Google application
+            credentials to use for the storage.
+        aws_access_key_id: Specify the AWS access key ID to use for the
+            storage.
+        aws_secret_access_key: Specify the AWS secret access key to use for the
+            storage.
+        aws_session_token: Specify the AWS session token to use for the
+            storage.
+        region_name: Specify the S3 region name to use for the storage.
+        s3_endpoint: Specify the S3 endpoint to use for the storage.
+    """
+
+    storage_type: str
+    label_config_type: str
+
+    prefix: Optional[str] = None
+    regex_filter: Optional[str] = ".*"
+    use_blob_urls: Optional[bool] = True
+    presign: Optional[bool] = True
+    presign_ttl: Optional[int] = 1
+    description: Optional[str] = ""
+
+    # credentials specific to the main cloud providers
+    azure_account_name: Optional[str]
+    azure_account_key: Optional[str]
+    google_application_credentials: Optional[str]
+    aws_access_key_id: Optional[str]
+    aws_secret_access_key: Optional[str]
+    aws_session_token: Optional[str]
+    region_name: Optional[str]
+    s3_endpoint: Optional[str]
+
+
+@step
+def get_or_create_dataset(
+    config: LabelStudioDatasetRegistrationConfig,
+    context: StepContext,
+) -> str:
+    """Gets preexisting dataset or creates a new one.
+
+    Args:
+        config: Step config.
+        context: Step context.
+
+    Returns:
+        The dataset name.
+
+    Raises:
+        StackComponentInterfaceError: If no active annotator could be found.
+    """
+    annotator = context.stack.annotator
```

Comment:
I'm guessing this needs a check like this:
```python
if not isinstance(annotator, LabelStudioAnnotator):
    raise TypeError()
```

Reply:
Same for the other steps below

Reply:
Good point. Fixed / done.